assets/js/dataclip-viewer/component.tsx

```diff
@@ -37,6 +39,21 @@ async function fetchDataclipContent(dataclipId: string) {
 const DataclipViewer = ({ dataclipId }: { dataclipId: string }) => {
   const [content, setContent] = useState<string>('');
 
+  const [monaco, setMonaco] = useState<Monaco | null>(null);
+  const [editor, setEditor] =
+    useState<__MonacoEditor.IStandaloneCodeEditor | null>(null);
+
+  useEffect(() => {
+    if (monaco && editor) {
+      addContextualCommand(
```

Comment:
So this is the actual fix? We override the keybinding and make it do nothing?

Can we maybe push this down into monaco.tx - the common component, and use a prop like `allowCommandPalette` to control it? Default false.

So most components, by default, will not allow you to open the command palette.

But the main editor should pass `allowCommandPalette` true, which allows it there.

I think that will just clean up some of the dependencies - you have to work quite hard to get the monaco references (and their types)

If we really need that context string, and we can't auto-generated it, can we pass it from the outside?

Reply:
@josephjclark would you mind giving me another round of review. Thanks man